GVFS/GVFS.Common/GVFSEnlistmentHealthCalculator.cs

```diff
@@ -0,0 +1,260 @@
+﻿using GVFS.Common.Database;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.Common
+{
+    /// <summary>
+    /// Class responsible for the business logic involved in calculating the health statistics
+    /// of a gvfs enlistment. Constructed with the lists of paths for the enlistment, and then
+    /// internally stores the calculated information. Compute or recompute via CalculateStatistics
+    /// with an optional parameter to only look for paths which are under the specified directory
+    /// </summary>
+    public class GVFSEnlistmentHealthCalculator
+    {
+        // In the context of this project, hydrated files are placeholders or modified paths
+        // The total number of hydrated files is this.PlaceholderCount + this.ModifiedPathsCount
+        private readonly GVFSEnlistmentPathData enlistmentPathData;
+
+        public GVFSEnlistmentHealthCalculator(GVFSEnlistmentPathData pathData)
+        {
+            this.enlistmentPathData = pathData;
+        }
+
+        public GVFSEnlistmentHealthData CalculateStatistics(string parentDirectory, int directoryOutputCount)
+        {
+            int gitTrackedItemsCount = 0;
+            int placeholderCount = 0;
+            int modifiedPathsCount = 0;
+            Dictionary<string, int> gitTrackedItemsDirectoryTally = new Dictionary<string, int>();

```

Comment:
Should we specify an explicit string comparison for the dictionary? The default string comparer is `Ordinal`

Reply:
Yeah I specified it as OrdinalIgnoreCase, the same as is used elsewhere in the health verb